state_test.go

```diff
@@ -41,6 +41,7 @@ type stateTestController struct {
 	state           State
 	updateResp      *client.UpdateResponse
 	updateRespErr   menderError
+	authorized      bool
 	authorize       menderError
```

Comment:
Since you are adding `authorized`, suggest renaming `authorize` to `authorizeErr` to avoid confusion.